File: WebAPI/BrewMapAPI/BrewMapAPI/Models/Drink.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace BrewMapAPI.Models
{
    public class Drink
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("name")]
        public string Name { get; set; }

        [BsonElement("description")]
        public string? Description { get; set; }

        [BsonElement("availableAtLocationId")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string AvailableAtLocationId { get; set; }

        [BsonElement("isVisible")]
        public bool IsVisible { get; set; } = true;

        [BsonElement("reportCount")]
        public int ReportCount { get; set; } = 0;

        [BsonElement("createdByUserId")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string CreatedByUserId { get; set; }

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("aggregatedRating")]
        public AggregatedRating AggregatedRating { get; set; } = new AggregatedRating();
    }

    public class AggregatedRating
    {
        [BsonElement("average")]
        public double Average { get; set; } = 0;

        [BsonElement("count")]
        public int Count { get; set; } = 0;
    }
}

File: WebAPI/MobileApp/app/src/main/java/hr/algebra/mobileapp/models/Drink.kt
package hr.algebra.mobileapp.models

/**
 * Mirrors `ReadDrink` DTO from the BrewMap API.
 *
 * `aggregatedRating` is kept as a nested object to match the JSON shape
 * returned by `GET api/Drink/{id}` and `GET api/Drink/location/{locationId}`.
 */
data class Drink(
    val id: String,
    val name: String,
    val description: String?,
    val availableAtLocationId: String,
    val createdByUserId: String,
    val createdAt: String,
    val updatedAt: String,
    val aggregatedRating: AggregatedRating
)

/**
 * Mirrors `ReadAggregatedRating` from the BrewMap API (used inside [Drink]).
 */
data class AggregatedRating(
    val average: Double,
    val count: Int
)

/**
 * Mirrors `ReadBestDrink` DTO — returned by
 * `GET api/Drink/location/{locationId}/best-drink`.
 */
data class BestDrink(
    val id: String,
    val name: String,
    val rating: Double
)
